### ISSUE TITLE
Unable to Retrieve Logs for Failed EMR Virtual Cluster Job Runs

### ISSUE THREAD

**Issue by AditModi** [awslabs/data-on-eks#140]
- I am facing an issue where my EMR virtual cluster job run is failing and I am unable to retrieve logs to debug the issue. When I try to open the logs, I get an error message saying "Internal failure in retrieving logs".

- I have tried checking the AWS EMR console for more details, but there is no additional information provided. I have also checked the CloudWatch logs, but they do not contain any logs related to the failed job run.

# Reproduction Code [Required]

## Steps to reproduce the behavior:

-  I followed the "EMR on EKS with Karpenter" guide to deploy the resources and then execute the sample Spark job to trigger compute optimized Karpenter provisioner. The job is supposed to be successful in execution and even if it fails, it's supposed to show where it fails which can be checked from the logs. However, I am unable to retrieve the logs due to the above-mentioned error.
    
# Expected behavior

I expected to be able to retrieve logs for the failed EMR virtual cluster job run.

# Actual behavior

I am unable to retrieve logs for the failed EMR virtual cluster job run.

**Comment by youngjeong46** [comment]
@AditModi Thanks for posting the issue. I'll reproduce the problem and get back to you.

**Comment by vara-bonthu** [comment]
You could try using `kubectl get events -n <namespace>` to see failures while initiating the EMR EKS Job pod. We have updated the blueprint recently so pull the latest. Thanks

**Comment by youngjeong46** [comment]
@AditModi, as @vara-bonthu suggested - I would pull the latest to use the latest blueprints. I was able to get logs in a variety of ways including:
1. Spark UI from the EMR console, looking at the Event Log, as well as logs at various stages and executor. On a failed job, this doesn't seem to be that useful.
2. `kubectl get events -n <namespace>` as mentioned above.
3. Looking at individual pod as they are executing: `kubectl logs -n emr-data-team-a taxidata-exec-1` using the example from the repo.

**Comment by AditModi** [comment]
@vara-bonthu @youngjeong46 ,

Thank you for your response and apologies for the delay in my reply.

I have tried using `kubectl get events -n <namespace>` to see failures while initiating the EMR EKS Job pod, but I am not able to find any relevant logs related to the failed job run.

I have also pulled the latest blueprints, but the issue still persists, and I am unable to retrieve any logs related to the failed EMR virtual cluster job run.

I have tried looking at the Spark UI from the EMR console, but as you mentioned, it doesn't seem to be that useful in case of failed jobs.

I will try looking at individual pod logs using `kubectl logs -n emr-data-team-a taxidata-exec-1` using the example from the repo and see if it helps me in debugging the issue.

Thank you for your help.

**Comment by AditModi** [comment]
@vara-bonthu @youngjeong46 ,

I wanted to let you know that I have been able to successfully resolve the issue I was experiencing with the failed EMR EKS Job pod.

After trying different approaches, I was able to retrieve the relevant logs using the kubectl logs -n emr-data-team-a taxidata-exec-1 command, as suggested by @youngjeong46 . This helped me in debugging the issue and identifying the root cause.

Thank you both for your guidance and support throughout this process. I really appreciate your help and expertise.
